packages/block-library/src/media-text/media-container.js

```diff
@@ -80,60 +53,84 @@ class MediaContainer extends Component {
 	}
 
 	renderPlaceholder() {
-		const { onSelectMedia, className } = this.props;
+		const {
+			mediaUrl,
+			mediaId,
+			mediaType,
+			toggleIsEditing,
+			onSelectMedia,
+			onSelectUrl,
+			className,
+		} = this.props;
+
+		const labels = {
+			title: ! mediaUrl ? __( 'Media' ) : __( 'Edit media' ),
+		};
+
+		const mediaPreview = ( !! mediaUrl && <img
+			alt={ __( 'Edit media' ) }
+			title={ __( 'Edit media' ) }
+			className={ 'edit-image-preview' }
+			src={ mediaUrl }
+		/> );
+
 		return (
 			<MediaPlaceholder
 				icon={ <BlockIcon icon={ icon } /> }
-				labels={ {
-					title: __( 'Media area' ),
-				} }
 				className={ className }
+				labels={ labels }
 				onSelect={ onSelectMedia }
+				onSelectURL={ onSelectUrl }
+				onDoubleClick={ toggleIsEditing }
+				onCancel={ !! mediaUrl && toggleIsEditing }
+				onError={ this.onUploadError }
 				accept="image/*,video/*"
 				allowedTypes={ ALLOWED_MEDIA_TYPES }
+				value={ { mediaId, mediaUrl } }
+				mediaPreview={ mediaType === 'image' && mediaPreview }
 			/>
 		);
 	}
 
 	render() {
-		const { mediaPosition, mediaUrl, mediaType, mediaWidth, commitWidthChange, onWidthChange } = this.props;
-		if ( mediaType && mediaUrl ) {
-			const onResize = ( event, direction, elt ) => {
-				onWidthChange( parseInt( elt.style.width ) );
-			};
-			const onResizeStop = ( event, direction, elt ) => {
-				commitWidthChange( parseInt( elt.style.width ) );
-			};
-			const enablePositions = {
-				right: mediaPosition === 'left',
-				left: mediaPosition === 'right',
-			};
+		const { isEditing, mediaPosition, mediaType, mediaWidth, commitWidthChange, onWidthChange } = this.props;
+		if ( isEditing ) {
```

Comment:
Being able to unwrap that if statement is a nice improvement to this method.

Reply:
thanks!